File: 2/2_2.php
<?php
declare(strict_types=1);

$forw = 0;
$aim = 0;
$depth = 0;

$fh = fopen('data.txt', 'r');
if ($fh) {
    while (!feof($fh)) {
        $line = fgets($fh);
        $n = $line ? trim($line) : null;
        if ($n === null) {
            continue;
        }

        [$direction, $amt] = explode(' ', $line);
        $amt = (int)$amt;

        switch ($direction) {
            case "down":
                $aim += $amt;
                break;
            case "up":
                $aim -= $amt;
                break;
            case "forward":
                $forw += $amt;
                $depthIncrease = $amt * $aim;
                $depth += $depthIncrease;
        }
    }
    fclose($fh);
}
var_dump($forw * $depth);

File: 2/2_2.py
#!/usr/bin/python3
from pprint import pprint

forw = 0
aim = 0
depth = 0

with open('data.txt') as file:
    for line in file:
        direction, amt = line.rstrip().split(" ")
        amt = int(amt)

        if direction == 'down':
            aim += amt
            continue
        if direction == 'up':
            aim -= amt
            continue
        if direction == 'forward':
            forw += amt
            depth_increase = amt * aim
            depth += depth_increase

pprint(forw * depth)
